### PR TITLE
Refs #5824 Topic Implementation. Minor updates to PSM API. [6613]

### PR DESCRIPTION
**Pull request by LuisGP**
* Adds Topic class.
* Adds TopicListener.
* PSM updates:
    * Implements `AnyTopic.hpp`.
    * Adds correct implementations to `detail/Status.hpp`.
* Other minor style fixes.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/663/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/575/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/658/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/139/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/665/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/577/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/660/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/141/)

File: include/dds/topic/detail/AnyTopic.hpp [hunk 1]
```diff
@@ -23,17 +23,14 @@
...
+#include <fastdds/dds/topic/Topic.hpp>
 
 namespace dds {
 namespace topic {
 namespace detail {
 
-//TODO: Fix when AnyTopicDelegate is implemented
-//typedef dds::topic::TAnyTopic<org::opensplice::topic::AnyTopicDelegate> AnyTopic;
-class AnyTopic { };
+typedef dds::topic::TAnyTopic<eprosima::fastdds::dds::Topic> AnyTopic;
```
Comment (laura-eprosima) [inline include/dds/topic/detail/AnyTopic.hpp]:
```suggestion
using AnyTopic = eprosima::fastdds::dds::Topic;
```

File: include/dds/core/status/detail/Status.hpp [hunk 2]
```diff
@@ -32,49 +37,38 @@ namespace core {
 namespace status {
 namespace detail {
 
-//TODO: Fix when StatusDelegate is implemented
-//typedef dds::core::status::TInconsistentTopicStatus< org::opensplice::core::InconsistentTopicStatusDelegate > InconsistentTopicStatus;
-class InconsistentTopicStatus { };
+using InconsistentTopicStatus =
+    dds::core::status::TInconsistentTopicStatus<eprosima::fastdds::dds::InconsistentTopicStatus>;
```
Comment (laura-eprosima) [inline include/dds/core/status/detail/Status.hpp]:
```suggestion
    eprosima::fastdds::dds::InconsistentTopicStatus;
```
Same for the rest

File: include/dds/core/status/Status.hpp [hunk 3]
```diff
@@ -440,31 +441,29 @@ class DataOnReadersStatus
...
+using RequestedDeadlineMissedStatus = dds::core::status::detail::RequestedDeadlineMissedStatus;
 
-typedef ::dds::core::status::detail::SampleLostStatus SampleLostStatus;
+using RequestedIncompatibleQosStatus = dds::core::status::detail::RequestedIncompatibleQosStatus;
 
-class StatusMask;
+using SampleLostStatus = dds::core::status::detail::SampleLostStatus;
 
-typedef ::dds::core::status::detail::SubscriptionMatchedStatus SubscriptionMatchedStatus;
+using SubscriptionMatchedStatus = dds::core::status::detail::SubscriptionMatchedStatus;
```
Comment (laura-eprosima) [inline include/dds/core/status/Status.hpp]: Remove the templates on the above classes and then you can remove all these aliases.

File: include/dds/core/status/Status.hpp [hunk 4]
```diff
@@ -36,8 +36,7 @@ namespace status {
 /**
  * \copydoc DCPS_Status_InconsistentTopic
  */
-template<typename D>
-class TInconsistentTopicStatus : public dds::core::Value<D>
+class TInconsistentTopicStatus : public dds::core::Value<detail::InconsistentTopicStatus>
```
Comment (laura-eprosima) [inline include/dds/core/status/Status.hpp]:
```suggestion
class InconsistentTopicStatus : public dds::core::Value<detail::InconsistentTopicStatus>
```
Same for the rest

File: include/dds/core/status/detail/Status.hpp [hunk 5]
```diff
@@ -21,11 +21,11 @@
 #include <fastdds/dds/core/status/BaseStatus.hpp>
 #include <fastdds/dds/core/status/PublicationMatchedStatus.hpp>
 #include <fastdds/dds/core/status/SubscriptionMatchedStatus.hpp>
-#include <fastrtps/qos/LivelinessChangedStatus.h>
-#include <fastrtps/qos/LivelinessLostStatus.h>
-#include <fastrtps/qos/DeadlineMissedStatus.h>
-#include <fastrtps/qos/IncompatibleQosStatus.hpp>
-#include <fastrtps/qos/SampleRejectedStatus.hpp>
+#include <fastdds/dds/qos/LivelinessChangedStatus.hpp>
+#include <fastdds/dds/core/status/BaseStatus.hpp>
```
Comment (laura-eprosima) [inline include/dds/core/status/detail/Status.hpp]: It is already included in line 21
```suggestion
```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/695/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/594/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/690/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/160/)

**Pull request merged**